spec/unit/backtracie_spec.rb

```diff
@@ -271,6 +271,118 @@ class ClassWithBlockInsideMethodDefinedUsingDefinedMethod
+            end
+            Object.send(:remove_const, :ToplevelModuleWithDefineMethodInIt)
+          RUBY
+
+          set_trace_func nil
+          bt
+        end
+      end
+
+      # These two function calls should never be reformatted to be on different lines!
+      # See above for a note on why this looks weird
+      let!(:backtraces_for_comparison) {
+        [test_method.call { described_class.backtrace_locations(Thread.current) }, test_method.call { Thread.current.backtrace_locations }]
+      }
+
+      it_should_behave_like "an equivalent of the Ruby API (using locations)"
+
+      it do
+        expect(backtracie_stack[3].qualified_method_name).to eq "Module\#{block}"
+      end
+    end
+
+    context "when sampling code inside a module" do
+      let(:test_method) do
+        proc do |&blk|
+          $backtracie_global_block = blk # rubocop:disable Style/GlobalVars
+          module CapturingBacktraceFromInsideAModule
+            $backtracie_bt = $backtracie_global_block.call # rubocop:disable Style/GlobalVars
+          end
+          $backtracie_bt # rubocop:disable Style/GlobalVars
+        end
+      end
+
+      # These two function calls should never be reformatted to be on different lines!
+      # See above for a note on why this looks weird
+      let!(:backtraces_for_comparison) {
+        [test_method.call { described_class.backtrace_locations(Thread.current) }, test_method.call { Thread.current.backtrace_locations }]
+      }
+
+      it_should_behave_like "an equivalent of the Ruby API (using locations)"
+
+      it do
+        expect(backtracie_stack[2].qualified_method_name).to eq "Module#<module:CapturingBacktraceFromInsideAModule>"
+      end
+    end
+
+    context 'when sampling a module extended hook' do
+      let(:test_method) do
+        proc do |&block|
+          $backtracie_global_block = block # rubocop:disable Style/GlobalVars
+
+          module ::ModuleToBeIncluded
+            def self.extended(other)
+              $backtracie_bt = $backtracie_global_block.call # rubocop:disable Style/GlobalVars
+            end
+          end
+
+          module ::ModuleToHaveIncludeApplied
+            extend ::ModuleToBeIncluded
+          end
+
+          Object.send(:remove_const, :ModuleToHaveIncludeApplied)
+          Object.send(:remove_const, :ModuleToBeIncluded)
+
+          $backtracie_bt # rubocop:disable Style/GlobalVars
+        end
+      end
+
+      # These two function calls should never be reformatted to be on different lines!
+      # See above for a note on why this looks weird
+      let!(:backtraces_for_comparison) {
+        [test_method.call { described_class.backtrace_locations(Thread.current) }, test_method.call { Thread.current.backtrace_locations }]
+      }
+
+      it_should_behave_like "an equivalent of the Ruby API (using locations)"
+
+      it do
+        expect(backtracie_stack[4].qualified_method_name).to eq "Module$singleton.<module:ModuleToHaveIncludeApplied>"
```

Comment:
And likewise, if you have a better way to represent this I'm all ears.

Reply:
[You are tearing me apart Ruby!](https://www.youtube.com/watch?v=IJ_icDmulqU)

I... think whatever "when sampling code inside a module" does, this one should do as well, since it's pretty much the same thing "I'm running code inside a module definition".

But like the above, let's leave this one as-is and revisit when we have a good idea for it :)